x-pack/plugins/maps/public/classes/sources/tms_source/index.ts

```diff
@@ -6,8 +6,15 @@
  */
 
 import { DataFilters } from '../../../../common/descriptor_types';
+import { DataRequest } from '../../util/data_request';
 import { ISource } from '../source';
-
+import { DataRequestMeta} from '../../../../common/descriptor_types';
+import { RasterTileSource } from '@kbn/mapbox-gl';
+export interface RasterTileSourceData {
+  url: string;
+}
 export interface ITMSSource extends ISource {
   getUrlTemplate(dataFilters: DataFilters): Promise<string>;
+  canSkipSourceUpdate(dataRequest:DataRequest,nextRequestMeta:DataRequestMeta): Promise<boolean>;
+  isSourceStale(mbSource:RasterTileSource,sourceDataRequest:object):boolean;
```

Comment:
Can you:
* create a type `TMSSourceData` that is `{ url?: string }`
* in isSourceStale method, change name of  `sourceDataRequest` parameter to `sourceData` and type as `TMSSourceData`
* In raster_tile_layer.ts, replace all usages of `{ url?: string }` with `TMSSourceData`

Reply:
I think that it might make sense to utilize the RasterTileSourceData interface that Nick started using to replace the {url?:string} since it is already defined.

Reply:
> I think that it might make sense to utilize the RasterTileSourceData interface that Nick started using to replace the {url?:string} since it is already defined.

Yes, that makes sense. I did not see this interface. Please use that.

Reply:
Can you move RasterTileSourceData interface to tms_source/index.ts?

Reply:
its already there :)